flang/lib/Semantics/check-omp-loop.cpp

```diff
@@ -262,6 +262,15 @@ void OmpStructureChecker::Enter(const parser::OpenMPLoopConstruct &x) {
     EnterDirectiveNest(SIMDNest);
   }
 
+  if (CurrentDirectiveIsNested() &&
+      llvm::omp::topTeamsSet.test(GetContext().directive) &&
+      GetContextParent().directive == llvm::omp::Directive::OMPD_target &&
+      !GetDirectiveNest(TargetBlockOnlyTeams)) {
```

Comment:
Not exactly relevant to this PR since a similar check is used somewhere else. But for the following example:
```
!$omp target ....
  !$omp teams ....
    !$omp target ....
      .... some statement ....
      !$omp teams ....
        ....
....
```

Would the semantic check trigger in this case? I think it won't since `GetDirectiveNest(TargetBlockOnlyTeams)` will return a non-zero result.

Reply:
>I am torn between spending time fixing it on the one hand while on the other hand letting what seems to be an obscure case slide.

No need to fix it, at least in this PR. Specially since the test is used somewhere else. The way the directive-nest test is currently implemented is just surprising to me (seems a bit too permisive) and wanted to verify if this is really the case.